pywbem/_server.py

```diff
@@ -299,6 +319,115 @@ def profiles(self):
             self._determine_profiles()
         return self._profiles
 
+    def create_namespace(self, namespace):
+        """
+        Create the specified CIM namespace in the WBEM server.
+
+        This method attempts the following approaches for creating the
+        namespace, in order, until an approach succeeds:
+
+        1. Namespace creation as described in the WBEM Server profile
+           (:term:`DSP1092`) via CIM method
+           `CIM_WBEMServer.CreateWBEMServerNamespace()`.
+
+        2. Issuing the `CreateInstance` operation on the `CIM_Namespace` class
+           against the Interop namespace. This is the approach described in
+           :term:`DSP0200`, except that pywbem targets the Interop namespace
+           instead of the namespace named "root".
+
+        Creating namespaces using the `__Namespace` pseudo-class has been
+        deprecated already in DSP0200 1.1.0 (released in 01/2003), and pywbem
+        does not implement that approach.
+
+        Parameters:
+
+            namespace (:term:`string`): CIM namespace name. Must not be `None`.
+              The namespace may contain leading and a trailing slash, both of
+              which will be ignored.
+
+        Returns:
+
+          :term:`unicode string`: The specified CIM namespace name in its
+          standard format (i.e. without leading or trailing slash characters).
+
+        Raises:
+
+            Exceptions raised by :class:`~pywbem.WBEMConnection`.
+            CIMError: CIM_ERR_ALREADY_EXISTS, Specified namespace already
+              exists in the WBEM server.
+            CIMError: CIM_ERR_NOT_FOUND, Interop namespace could not be
+              determined.
+            CIMError: CIM_ERR_NOT_FOUND, Unexpected number of
+              `CIM_ObjectManager` instances.
+            CIMError: CIM_ERR_FAILED, Unexpected number of
+              central instances of WBEM Server profile.
+        """
+
+        std_namespace = _ensure_unicode(namespace.strip('/'))
+
+        # Try approach 1: Method defined in WBEM Server profile
+
+        ws_profiles = self.get_selected_profiles('DMTF', 'WBEM Server')
```

Comment:
As long as we are getting the WBEM_Server instance here through the profile, should we consider making that public or its access public because it will be used for other things than just creating namespace???

Reply:
Either the result or the method to get it.

Reply:
No. Let's not do that before we understand the use cases. This is a MOF compiler API and the user supplies a target repository connection, so the user already has everything they need to set up their own WBEM_Server if they need one. Just saving a few operations is not worth the function creep.